src/cdk-experimental/combobox/combobox.spec.ts

```diff
@@ -69,21 +141,290 @@ describe('Combobox', () => {
+      testComponent.actions = 'downKey';
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeFalse();
+
+      dispatchKeyboardEvent(comboboxElement, 'keydown', DOWN_ARROW);
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeTrue();
+    });
+
+    it('should toggle panel with toggle open action', () => {
+      testComponent.actions = 'toggle';
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeFalse();
+
+      dispatchMouseEvent(comboboxElement, 'click');
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeTrue();
+
+      dispatchMouseEvent(comboboxElement, 'click');
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeFalse();
+    });
+
+    it('should close panel on escape key', () => {
+      testComponent.actions = 'click';
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeFalse();
+
+      dispatchMouseEvent(comboboxElement, 'click');
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeTrue();
+
+      dispatchKeyboardEvent(comboboxElement, 'keydown', ESCAPE);
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeFalse();
+    });
+
+    it('should handle multiple open actions', () => {
+      testComponent.actions = 'click downKey';
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeFalse();
+
+      dispatchMouseEvent(comboboxElement, 'click');
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeTrue();
+
+      dispatchKeyboardEvent(comboboxElement, 'keydown', ESCAPE);
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeFalse();
+
+      dispatchKeyboardEvent(comboboxElement, 'keydown', DOWN_ARROW);
+      fixture.detectChanges();
+
+      expect(comboboxInstance.isOpen()).toBeTrue();
+    });
+  });
 });
 
 @Component({
   template: `
   <button cdkCombobox #toggleCombobox class="example-combobox"
           [cdkComboboxTriggerFor]="panel"
-          [openActions]="'focus'">
+          [openActions]="actions">
     No Value
   </button>
-
+  <div id="other-content"></div>
   <ng-template cdkComboboxPanel #panel="cdkComboboxPanel">
-    Panel Content
+    <div dialogContent #dialog="dialogContent" [parentPanel]="panel">
+      <input #input>
+      <button id="applyButton" (click)="panel.closePanel(input.value)">Apply</button>
+    </div>
   </ng-template>`,
 })
 class ComboboxToggle {
+  @ViewChild('input') inputElement: ElementRef<HTMLInputElement>;
+
+  actions: string = 'click';
+}
+
+export const PANEL = new InjectionToken<CdkComboboxPanel>('CdkComboboxPanel');
+
+let id = 0;
+
+@Directive({
+  selector: '[dialogContent]',
+  exportAs: 'dialogContent',
+  host: {
+    '[attr.role]': 'role',
+    '[id]': 'dialogId',
+    'tabIndex': '-1'
+  }
+})
+export class DialogContent<V> implements OnInit {
```

Comment:
I might make this `FakeDialogContent` just so it doesn't come up for IDE shortcuts like shift+shift when looking for dialog stuff